src/sentry/search/events/builder.py

```diff
@@ -2093,6 +2094,23 @@ def get_default_value(meta_type: str) -> Any:
             return None
 
 
+class AlertMetricsQueryBuilder(MetricsQueryBuilder):
+    def __init__(
+        self,
+        *args: Any,
+        granularity: int,
+        **kwargs: Any,
+    ):
+        self._granularity = granularity
+        super().__init__(*args, **kwargs)
+
+    def resolve_limit(self, limit: Optional[int]) -> Optional[Limit]:
+        return None
+
+    def resolve_granularity(self) -> Granularity:
+        return Granularity(self._granularity)
+
+
```

Comment:
@wmak This looks a lot cleaner, good call

Reply:
Niiice :+1: